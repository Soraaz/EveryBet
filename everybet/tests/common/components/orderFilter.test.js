import React from "react";
import {Platform} from "react-native";
import {shallow} from "enzyme";

import OrderFilter from "../../../src/common/components/orderFilter";
import renderer from "react-test-renderer";

describe("OrderFilter", () => {
	describe("rendering", () => {
		/*		describe("android", () => {
					beforeAll(() => {
						Platform.OS = "android";
					});

					it("Should render correctly", () => {
						const tree = renderer
							.create(<OrderFilter filterCallback={() => {}}/>)
							.toJSON();
						expect(tree).to.matchSnapshot();
					});
				});
		*/
		describe("ios", () => {
			beforeAll(() => {
				Platform.OS = "ios";
			});

			it("Should render correctly", () => {
				const tree = renderer
					.create(<OrderFilter filterCallback={() => {}}/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});
		});
	});

	describe("functionalities", () => {
		it("Should set state on mount", (done) => {
			const wrapper = shallow(<OrderFilter filterCallback={() => {}}/>);
			process.nextTick(() => {
				wrapper.update();
				expect(wrapper.state("orderList").length).to.equal(3);
				done();
			});
		});

		it("Should set state on change order", (done) => {
			const wrapper = shallow(<OrderFilter filterCallback={() => {}}/>);
			wrapper.instance()._changeOrder("Populaires");
			process.nextTick(() => {
				wrapper.update();
				expect(wrapper.state("order")).to.equal("Populaires");
				done();
			});
		});
	});
});
