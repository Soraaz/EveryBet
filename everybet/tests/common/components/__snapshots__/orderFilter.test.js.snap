// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderFilter rendering ios Should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "marginHorizontal": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
      }
    }
  >
    Ordre
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 5,
        "height": 25,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingLeft": 10,
        }
      }
    >
      <Text>
        Plus r√©cents
      </Text>
    </View>
  </View>
</View>
`;
